catroid/src/main/java/org/catrobat/catroid/formulaeditor/FormulaEditorHistory.java

```diff
@@ -73,11 +82,11 @@ public InternFormulaState forward() {
 	}
 
 	public void updateCurrentSelection(InternFormulaTokenSelection internFormulaTokenSelection) {
-		current.setSelection(internFormulaTokenSelection);
+		current.internFormulaState.setSelection(internFormulaTokenSelection);
```

Comment:
Also here: What if current.internFormulaState == null? Is there a way to provide a function to directly set the selection?